src/EditorFeatures/CSharpTest/UseAutoProperty/UseAutoPropertyTests.cs

```diff
@@ -1403,8 +1383,189 @@ protected set
 }",
 @"class Class
 {
-
     public int P { get; protected set; }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseAutoProperty)]
+        [WorkItem(26858, "https://github.com/dotnet/roslyn/issues/26858")]
+        public async Task TestPreserveTrailingTrivia1()
+        {
+            await TestInRegularAndScriptAsync(
+@"class Goo
+{
+    private readonly object [|bar|] = new object();
+
+    public object Bar => bar;
+    public int Baz => 0;
+}",
+@"class Goo
+{
+    public object Bar { get; } = new object();
+    public int Baz => 0;
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseAutoProperty)]
+        [WorkItem(26858, "https://github.com/dotnet/roslyn/issues/26858")]
+        public async Task TestPreserveTrailingTrivia2()
+        {
+            await TestInRegularAndScriptAsync(
+@"class Goo
+{
+    private readonly object [|bar|] = new object();
+
+    public object Bar => bar; // prop comment
+    public int Baz => 0;
+}",
+@"class Goo
+{
+    public object Bar { get; } = new object(); // prop comment
+    public int Baz => 0;
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseAutoProperty)]
+        [WorkItem(26858, "https://github.com/dotnet/roslyn/issues/26858")]
+        public async Task TestPreserveTrailingTrivia3()
+        {
+            await TestInRegularAndScriptAsync(
+@"class Goo
+{
+    private readonly object [|bar|] = new object();
+
+    // doc
+    public object Bar => bar; // prop comment
+    public int Baz => 0;
+}",
+@"class Goo
+{
+    // doc
+    public object Bar { get; } = new object(); // prop comment
+    public int Baz => 0;
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseAutoProperty)]
+        [WorkItem(26858, "https://github.com/dotnet/roslyn/issues/26858")]
+        public async Task TestKeepLeadingBlank()
+        {
+            await TestInRegularAndScriptAsync(
+@"class Goo
+{
+
+    private readonly object [|bar|] = new object();
+
+    // doc
+    public object Bar => bar; // prop comment
+    public int Baz => 0;
+}",
+@"class Goo
+{
+
+    // doc
+    public object Bar { get; } = new object(); // prop comment
+    public int Baz => 0;
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseAutoProperty)]
+        public async Task TestMultipleFieldsAbove1()
+        {
+            await TestInRegularAndScriptAsync(
+@"class Class
+{
+    [|int i|];
+    int j;
+
+    int P
+    {
+        get
+        {
+            return i;
+        }
+    }
+}",
+@"class Class
+{
+    int j;
+
+    int P { get; }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseAutoProperty)]
+        public async Task TestMultipleFieldsAbove2()
```

Comment:
>TestMultipleFieldsAbove2 [](start = 26, length = 24)

Consider another variant of this test, where there is an empty line between `int j` and `int i`.
The question is what happens to the empty line above `int i` and the empty line between `int i` and the property for `i`. 

I'd expect we end up with a single empty line between `int j` (first field) and the property. Is that correct?

Reply:
Will do!

Reply:
@jcouv Let me know if the tests are sufficient :)